#Annotation and Visualisation of RNA-seq

# Overview

- Visualising DE results
- Getting annotation
- Retrieving gene models
- Exporting browser tracks
- Visualising results with respect to genomic location

We have a list of significantly differentially expressed genes, but the only annotation we can see is the Entrez Gene ID, which is not very informative. 

```{r}
results <- as.data.frame(topTags(lrt.BvsL,n = Inf))
results
dim(results)
```

`edgeR` provides a function `plotSmear` that allows us to visualise the results of a DE analysis. In a similar manner to the [*MA-plot* for microarray data](https://en.wikipedia.org/wiki/MA_plot), this plot shows the log-fold change against log-counts per million, with DE genes highlighted:

```{r}
summary(de <- decideTestsDGE(lrt.BvsL))
detags <- rownames(dgeObj)[as.logical(de)]
plotSmear(lrt.BvsL, de.tags=detags)
```

To perform such a task we need to map between the identifiers we have in the `edgeR` output and more familiar names.

Finally, we will look at sophisticated visualisations that allow us to incorporate information about the structure of a gene, level of sequencing coverage.

## Adding Annotation to the edgeR Results

There are a number of ways to add annotation, but we will demonstrate how to do this using the *org.Mm.eg.db* package. This package is one of several *organism-level* packages which are re-built every 6 months.

```{r message=FALSE}
library(org.Mm.eg.db)
```

First, decide what information is needed. Use the `columns` function on the annotation database to see available data.

```{r}
columns(org.Mm.eg.db)
```

Retrieve gene symbols and full gene names using the `select` function:

```{r}
ann <- select(org.Mm.eg.db,keys=rownames(results),columns=c("ENTREZID","SYMBOL","GENENAME"))
head(ann)
```

Bind the annotation information to the `results` data frame:

```{r}
results.annotated <- cbind(results, ann)
head(results.annotated)
```

Save the results table:

```{r}
write.csv(results.annotated,file="B.PregVsLacResults.csv",row.names=FALSE)
```

## Visualisation

### Volcano Plot

```{r,fig.height=5,fig.width=10}
signif <- -log10(results.annotated$FDR)
plot(results.annotated$logFC,signif,pch=16)
points(results.annotated[detags,"logFC"],-log10(results.annotated[detags,"FDR"]),pch=16,col="red")
```

### Stripchart for Individual Genes

```{r,fig.width=12,fig.height=5}
library(RColorBrewer)
normCounts <- dgeObj$counts
nice.col <- brewer.pal(6,name="Dark2")
stripchart(normCounts["110308",]~group,vertical=TRUE,las=2,cex.axis=0.8,pch=16,cex=1.3,col=nice.col,method="jitter",ylab="Normalised log2 expression",main="Krt5")
```

## Exporting Browser Tracks

Create a `.bed` track for DE genes:

```{r}
library(rtracklayer)
mcols(sigRegions)$score <- -log10(sigRegions$FDR)
export(sigRegions , con = "topHits.bed")
```

## Retrieving Genomic Locations

Use `TxDb.Mmusculus.UCSC.mm10.knownGene` to retrieve exon coordinates for genes:

```{r message=FALSE}
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
tx <- TxDb.Mmusculus.UCSC.mm10.knownGene
keys <- c("50916","110308","12293")
genePos <- select(tx, keys=keys, keytype="GENEID", columns=c("EXONCHROM","EXONSTART","EXONEND"))
head(genePos)
```

## Summary

This report outlines differential expression analysis and genomic visualisation using R. Exported tracks and genomic annotations provide insights into DE genes in mouse mammary tissue.
