#RNA Preprocessing

## Data Files and Setup

- Data files:  
  - `sampleinfo.txt`  
  - `GSE60450_Lactation-GenewiseCounts.txt`  
  - `mouse_c2_v5.rdata`  
  - `mouse_H_v5.rdata`  
- Data files are available from: [https://figshare.com/s/1d788fd384d33e913a2a](https://figshare.com/s/1d788fd384d33e913a2a). Place them in the `/data` directory.

## Packages Used

```{r setup, message = FALSE}
library(edgeR)
library(limma)
library(Glimma)
library(gplots)
library(org.Mm.eg.db)
library(RColorBrewer)
```

## Overview

- Reading in table of counts
- Filtering lowly expressed genes
- Quality control
- Normalization for composition bias

## Introduction

RNA-Seq technology allows genome-wide gene expression measurement and analysis. This report outlines the analysis steps starting from count data, proceeding through filtering, quality control, normalization, and visualization.

## Data Import

### Reading Sample Information

```{r loadSampleInfo}
sampleinfo <- read.delim("data/sampleinfo.txt")
View(sampleinfo)
sampleinfo
```

### Reading Count Data

```{r loadData}
seqdata <- read.delim("data/GSE60450_Lactation-GenewiseCounts.txt", stringsAsFactors = FALSE)
head(seqdata)
dim(seqdata)
```

### Formatting the Data

```{r formatData}
countdata <- seqdata[, -(1:2)]
rownames(countdata) <- seqdata[, 1]
colnames(countdata) <- substr(colnames(countdata), 1, 7)
all(colnames(countdata) == sampleinfo$SampleName)
```

## Filtering Lowly Expressed Genes

```{r filterGenes}
myCPM <- cpm(countdata)
thresh <- myCPM > 0.5
keep <- rowSums(thresh) >= 2
counts.keep <- countdata[keep, ]
```

## Converting Counts to DGEList Object

```{r makeDGEObj}
dgeObj <- DGEList(counts.keep)
dgeObj$samples
```

## Quality Control

### Library Sizes

```{r plotLibrarySizes}
barplot(dgeObj$samples$lib.size, names = colnames(dgeObj), las = 2, main = "Barplot of Library Sizes")
abline(h = 20e6, lty = 2)
```

### Distribution of Log-CPM

```{r plotLogCounts}
logcounts <- cpm(dgeObj, log = TRUE)
boxplot(logcounts, xlab = "", ylab = "Log2 counts per million", las = 2)
abline(h = median(logcounts), col = "blue")
```

### Multidimensional Scaling (MDS) Plot

```{r plotMDS}
col.cell <- c("purple", "orange")[sampleinfo$CellType]
col.status <- c("blue", "red", "dark green")[sampleinfo$Status]
par(mfrow = c(1, 2))
plotMDS(dgeObj, col = col.cell, main = "Cell Type")
legend("topleft", fill = c("purple", "orange"), legend = levels(sampleinfo$CellType))
plotMDS(dgeObj, col = col.status, main = "Status")
legend("topleft", fill = c("blue", "red", "dark green"), legend = levels(sampleinfo$Status))
```

## Normalization for Composition Bias

```{r calcNormFactors}
dgeObj <- calcNormFactors(dgeObj)
dgeObj$samples
```

### Mean-Difference Plots Before and After Normalization

```{r plotMD}
par(mfrow = c(1, 2))
plotMD(logcounts, column = 7)
abline(h = 0, col = "grey")
plotMD(dgeObj, column = 7)
abline(h = 0, col = "grey")
```

## Save Data for Future Use

```{r saveData}
save(dgeObj, sampleinfo, file = "Robjects/preprocessing.Rdata")
```

## References

- Fu, N. et al. (2015). Nature Cell Biology. [PMID: 25730472](http://www.ncbi.nlm.nih.gov/pubmed/25730472)
- Robinson, M. D., McCarthy, D. J., & Smyth, G. K. (2010). edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. *Bioinformatics*, 26(1), 139-140.
